### PR TITLE
[AAQ-486] add gemini option

### PR DESCRIPTION
**Pull request by suzinyou**
Reviewer: @amiraliemami 
Estimate: 30min

---

## Ticket

Fixes: https://idinsight.atlassian.net/browse/AAQ-486

### Goal

Add option to use Gemini with LiteLLM proxy server

### Changes

1. New Gemini API key secret
2. New parameters to disable Gemini's default safety checks which don't work with our guardrails
3. Commented out config for using Gemini in litellm proxy config

Also renamed aws secret open_ai -> openai

## Checklist

Fill with `x` for completed. Delete any lines that are not relevant

- [x] My code follows the style guidelines of this project
- [x] I have reviewed my own code to ensure good quality
- [x] I have tested the functionality of my code to ensure it works as intended
- [x] I have resolved merge conflicts
- [ ] I have updated the automated tests (if applicable)
- [ ] I have updated the requirements (if applicable)
- [ ] I have updated the README file (if applicable)
- [ ] I have updated affected documentation (if applicable)
- [ ] I have added blogpost in Latest Updates

### CODE REVIEW

File: deployment/aws/infrastructure/demo/main/credentials.tf [hunk 1]
```diff
@@ -106,18 +106,36 @@ resource "aws_secretsmanager_secret_version" "whatsapp_verify_token_secret" {
...
 
-resource "aws_secretsmanager_secret_version" "open_ai_key_secret" {
-  # The secret version is created for the open ai key secret.
+resource "aws_secretsmanager_secret_version" "openai_key_secret" {
+  # The secret version is created for the OpenAI API key secret.
   # The value will be added manually to the secret version.
-  secret_id     = aws_secretsmanager_secret.open_ai_key_secret.id
-  secret_string = "placeholder"
+  secret_id     = aws_secretsmanager_secret.openai_key_secret.id
+  secret_string = "replace"
```
Comment (amiraliemami) [inline deployment/aws/infrastructure/demo/main/credentials.tf:127]: let's call this placeholder? It's clearer

File: deployment/aws/infrastructure/testing/main/credentials.tf [hunk 2]
```diff
@@ -106,18 +106,36 @@ resource "aws_secretsmanager_secret_version" "whatsapp_verify_token_secret" {
...
 
-resource "aws_secretsmanager_secret_version" "open_ai_key_secret" {
-  # The secret version is created for the open ai key secret.
+resource "aws_secretsmanager_secret_version" "openai_key_secret" {
+  # The secret version is created for the OpenAI API key secret.
   # The value will be added manually to the secret version.
-  secret_id     = aws_secretsmanager_secret.open_ai_key_secret.id
-  secret_string = "placeholder"
+  secret_id     = aws_secretsmanager_secret.openai_key_secret.id
+  secret_string = "replace"
```
Comment (amiraliemami) [inline deployment/aws/infrastructure/testing/main/credentials.tf:127]: same here and below

File: core_backend/app/llm_call/utils.py [hunk 3]
```diff
@@ -36,6 +36,24 @@ async def _ask_llm_async(
         api_base=litellm_endpoint,
         api_key=LITELLM_API_KEY,
         metadata={"generation_name": litellm_model},
+        safety_settings=[
```
Comment (amiraliemami) [inline core_backend/app/llm_call/utils.py:39]: this now always gets sent regardless of which model the proxy routes to right? Could you add a comment in the code clarifying this is only for gemini maybe?

**Review by amiraliemami — APPROVED** [review]
Nice, thanks! Approving to unblock but let's co-work so I can get the API key and test it.

File: core_backend/app/llm_call/utils.py (hunk 3 above)
Comment (suzinyou) [reply core_backend/app/llm_call/utils.py:39]: Good idea thanks!

Comment (amiraliemami) [reply core_backend/app/llm_call/utils.py:40]: OpenAI errors out if this is sent to it. Should move this to the proxy definition

**Pull request merged**